File: ecommerc_dotnet/application/dto/DeliveryDto.cs
using System.ComponentModel.DataAnnotations;

namespace ecommerc_dotnet.dto
{
    public class DeliveryDto
    {
        public Guid Id { get; set; }
        public Guid UserId { get; set; }
        public bool IsAvailable { get; set; } = true;
        public DateTime? UpdatedAt { get; set; } = null;
        public string? Thumbnail { get; set; }
        public DeliveryAddressDto? Address { get; set; } = null;
        public DeliveryAnalysDto? Analys { get; set; } = null;
        public UserDeliveryInfoDto? User { get; set; } = null;
    }


    public class DeliveryAnalysDto
    {
        public decimal? DayFee { get; set; } = null;
        public decimal? WeekFee { get; set; } = null;
        public decimal? MonthFee { get; set; } = null;
        public int? DayOrder { get; set; } = null;
        public int? WeekOrder { get; set; } = null;
    }

    public class CreateDeliveryDto
    {
        [Required] public Guid UserId { get; set; }
        public IFormFile? Thumbnail { get; set; } = null;
    }
    
    public class UpdateDeliveryDto
    {
        public IFormFile? Thumbnail { get; set; } = null;
        public decimal? Longitude { get; set; } = null;
        public decimal? Latitude { get; set; } = null;

        [StringLength(maximumLength: 50 , ErrorMessage = "Enter Valide Name")]
        public string? Name { get; set; } = null;
        
        [StringLength(maximumLength: 13, ErrorMessage = "Enter Valide Name")]
        public string? Phone { get; set; } = null;
        public IFormFile? UserThumbnail { get; set; } = null;
        public string? Password { get; set; } = null;
        public string? NewPassword { get; set; } = null;
    }
}

File: ecommerce-delivery-man/app/src/main/java/com/example/e_commerc_delivery_man/dto/DeliveryDto.kt
package com.example.e_commerc_delivery_man.dto

import com.example.e_commerc_delivery_man.dto.UserDto
import com.example.hotel_mobile.services.kSerializeChanger.UUIDKserialize
import kotlinx.serialization.Serializable
import java.io.File
import java.util.UUID

@Serializable
data class DeliveryDto(
    @Serializable(with = UUIDKserialize::class)
    val id: UUID,
    @Serializable(with = UUIDKserialize::class)
    val userId: UUID,
    val isAvailable: Boolean,
    val thumbnail: String? = null,
    val address: AddressDto,
    val user: UserDto,
    val analys:DeliveryAnalysDto?=null
)

@Serializable()
data class DeliveryAnalysDto(
    val dayFee: Double? = null,
    val weekFee: Double? = null,
    val monthFee: Double? = null,
    val dayOrder: Int? = null,
    val weekOrder: Int? = null
)


